repose-aggregator/core/repose-core/src/main/java/org/openrepose/nodeservice/request/RequestHeaderServiceImpl.java

```diff
@@ -118,7 +122,7 @@ public void setVia(HttpServletRequestWrapper request) {
         @Override
         public void configurationUpdated(ContainerConfiguration configurationObject) {
             if (configurationObject.getDeploymentConfig() != null) {
-                viaReceivedBy = configurationObject.getDeploymentConfig().getVia();
+                viaReceivedBy = containerConfigurationService.getVia().orElse(null);
```

Comment:
As @adrianjgeorge pointed out, this is technically broken. The order of updates is indeterminate, so this may just be re-applying the current config. Most of the time, it will catch up, but it's still wrong.